xml2/MQMQA.rng

```diff
@@ -1,32 +1,162 @@
+            </element>
         </element>
     </define>
     <define name="MQMQA.model" combine="choice">
         <element name="Model">
             <attribute name="type">
+                <a:documentation>Modified Quasichemical Model in the Quadruplet Approximation</a:documentation>
                 <value>MQMQA</value>
             </attribute>
-            <optional>
-                <text/>
-            </optional>
+            <interleave>
+                <ref name="MQMConstituentArray"/>
+                <element name="ChemicalGroups">
+                    <a:documentation>Mapping of catipn/anion species to a integer chemical groups. Equal integers mean the species belong to the same group.</a:documentation>
+                    <interleave>
+                        <element name="Cations">
+                            <oneOrMore>
+                                <element name="Constituent">
+                                    <attribute name="groupid">
+                                        <data type="integer"/>
+                                    </attribute>
+                                    <attribute name="refid">
+                                        <data type="string"/>
+                                    </attribute>
+                                </element>
+                            </oneOrMore>
+                        </element>
+                        <element name="Anions">
+                            <oneOrMore>
+                                <element name="Constituent">
+                                    <attribute name="groupid">
+                                        <data type="integer"/>
+                                    </attribute>
+                                    <attribute name="refid">
+                                        <data type="string"/>
+                                    </attribute>
+                                </element>
+                            </oneOrMore>
+                        </element>
+                    </interleave>
+                </element>
+            </interleave>
         </element>
         <zeroOrMore>
             <element name="Parameter">
                 <attribute name="type">
-                    <value>Q</value>
+                    <choice>
+                        <value>MQMG</value><a:documentation>Gibbs energy of a pair</a:documentation>
+                        <value>MQMZ</value><a:documentation>Coordination number of a quadruplet. Coordination numbers for pairs must be given, default coordination numbers for other quadruplets can be generated automatically. Coordination numbers must result in a charge neutral quadruplet.</a:documentation>
+                        <value>MQMX</value><a:documentation>Excess Gibbs energy for a quadruplet</a:documentation>
+                    </choice>
                 </attribute>
+                <!--TODO: can the different parameter types be part of the <choice> tag?"-->
```

Comment:
It might help for the parameter specific options to have a `<choice>` of `<group>` so that the optional parameter-specific tags are mutually exclusive. 

As an alternative, maybe there needs to be a `<choice>` of `<element name=“Parameter”>` tags, instead of allowing types to be mixed-and-matched with different specialty options (like MQMZ and the Exponents element, which would be wrong).